miniapp/io.cpp

```diff
@@ -389,6 +409,64 @@ std::ostream& operator<<(std::ostream& o, const cl_options& options) {
     return o;
 }
 
+
+/// Parse spike times from a stream
+/// A single spike per line, white space is ignored
+/// All characters after a parsed floating point nr on a line are ignored
+/// Throws a usage error when parsing fails
+///
+/// Returns a unique_ptr to a vector of time_type
+
+std::unique_ptr<std::vector<time_type> > parse_spike_times_from_stream(
+    std::ifstream & fid) {
+    std::vector<time_type> * times = new std::vector<time_type>();
+    std::string line;
+    for (unsigned idx=0; std::getline(fid, line); ++idx) {
+        // Remove all white space from the line
+        line.erase(remove_if(line.begin(), line.end(), isspace), line.end());
+        // Continue if we have empty line
+        if (line.size() == 0) {
+            continue;
+        }
+
+        //make a stream from the line
+        std::istringstream in(line);
+        time_type parsed_value;
+        // Attempt to read
+        in >> parsed_value;
+
+        // When we encounter a problem
+        if (in.fail()) {
+            std::string error_str = std::string("Unable to parse file with input spikes! \n")
+                + "Line #:" + std::to_string(idx) + "\n"
+                + "Problematic content: " + line + "\n";
+            throw usage_error(error_str.c_str());
+        }
+        times->push_back(parsed_value);
+    }
+
+    return  std::unique_ptr<std::vector<time_type> >(std::move(times));
+}
+
+/// Parse spike times from a file supplied in path
+/// A single spike per line, white space is ignored
+/// All characters after a parsed floating point nr. on a line are ignored
+/// Throws a usage error when opening file or parsing fails
+///
+/// Returns a unique_ptr to a vector of time_type
+
+std::unique_ptr<std::vector<time_type> > get_parsed_spike_times_from_path(
+    const std::string& path) {
```

Comment:
Done

The eating of the first character does not happen when I emit a string with strprintf on this location... interesting